# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/TextEditSubActionHandler.cpp
# Commit: cdcef4f3f635
# Full Hash: cdcef4f3f63553a28a1055e6f4f051778ca8fb0a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-09-22 09:56:47
# Regressor Bug: 1789967
# File Overlap Count: 1
# Description:
#   Bug 1789967 - part 2: Make `TextEditor` and `HTMLEditor` implement `EditorBase::CollapseSelectionToEndOfLastLeafNode` by themselves r=m_kato
#   
#   It does different thing for `TextEditor` and `HTMLEditor`, and used almost
#   internally.  Therefore, it should be implemented in the sub classes and
#   we should name them better.
# ==============================================================================

diff -r cc2e8b7c548e -r cdcef4f3f635 editor/libeditor/TextEditSubActionHandler.cpp
--- a/editor/libeditor/TextEditSubActionHandler.cpp	Thu Sep 22 06:06:54 2022 +0000
+++ b/editor/libeditor/TextEditSubActionHandler.cpp	Thu Sep 22 06:17:36 2022 +0000
@@ -271,16 +271,16 @@
     return NS_OK;
   }
 
-  nsresult rv = CollapseSelectionToEndOfLastLeafNode();
+  nsresult rv = CollapseSelectionToEndOfTextNode();
   if (MOZ_UNLIKELY(rv == NS_ERROR_EDITOR_DESTROYED)) {
     NS_WARNING(
-        "EditorBase::CollapseSelectionToEndOfLastLeafNode() caused destroying "
-        "the editor");
+        "TextEditor::CollapseSelectionToEndOfTextNode() caused destroying the "
+        "editor");
     return NS_ERROR_EDITOR_DESTROYED;
   }
   NS_WARNING_ASSERTION(
       NS_SUCCEEDED(rv),
-      "EditorBase::CollapseSelectionToEndOfLastLeafNode() failed, but ignored");
+      "TextEditor::CollapseSelectionToEndOfTextNode() failed, but ignored");
 
   return NS_OK;
 }